beacon-chain/state/state-native/v1/beacon_state_mainnet.go

```diff
@@ -0,0 +1,50 @@
+// +build !minimal
+
+package v1
+
+import (
+	"sync"
+
+	eth2types "github.com/prysmaticlabs/eth2-types"
+	"github.com/prysmaticlabs/go-bitfield"
+	customtypes "github.com/prysmaticlabs/prysm/beacon-chain/state/state-native/custom-types"
+	"github.com/prysmaticlabs/prysm/beacon-chain/state/state-native/fieldtrie"
+	"github.com/prysmaticlabs/prysm/beacon-chain/state/stateutil"
+	"github.com/prysmaticlabs/prysm/beacon-chain/state/types"
+	ethpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
+)
+
+// BeaconState defines a struct containing utilities for the Ethereum Beacon Chain state, defining
+// getters and setters for its respective values and helpful functions such as HashTreeRoot().
+type BeaconState struct {
+	genesisTime                 uint64                      `ssz-gen:"true"`
+	genesisValidatorsRoot       customtypes.Byte32          `ssz-gen:"true" ssz-size:"32"`
+	slot                        eth2types.Slot              `ssz-gen:"true"`
+	fork                        *ethpb.Fork                 `ssz-gen:"true"`
+	latestBlockHeader           *ethpb.BeaconBlockHeader    `ssz-gen:"true"`
+	blockRoots                  *customtypes.BlockRoots     `ssz-gen:"true" ssz-size:"8192,32"`
```

Comment:
I wish we had a better way of putting these values somewhere else than hard-coding them. Maybe using go build tags and doing some magic? Unsure about this, but it's my main concern